docs: Update docs with ThemeBuilder changes (#1402)
unoplatform/Uno.Themes: doc/material-dsp.md

* The package is already present when you select _Material_ theme during project creation:
   ![Selection of Material theme when creating a project using the Uno Template Wizard](assets/material-theme-selection-wizard.png)

> ## Generating a custom color palette and export as JSON file

Reviewer: ```suggestion
## Generating a custom color palette and exporting as a JSON file
```

> ## Generating a custom color palette and exporting as a JSON file

Reviewer 2: ```suggestion
## Generating a custom color palette and exporting it as a JSON file
```

1. Navigate to the [Material Theme Builder](https://m3.material.io/theme-builder#/custom) and select the colors you want to use for your application.
2. Locate the _Export_ button and pick the _Material Theme (JSON)_ format.

   ![Export Button](assets/material-theme-builder-export1.png) ![DSP Selection](assets/material-theme-builder-export2.png)
3. Save the file to your computer.

> 4. Replace the file `ColorPaletteOverride.json` in the `Styles` folder of your application project with the one you just downloaded.

Reviewer 2: ```suggestion
4. Replace the `ColorPaletteOverride.json` file in the `Styles` folder of your application project with the one you just downloaded.
```

5. Build your application. The `ColorPaletteOverride.xaml` file will be automatically updated with the colors present in the JSON file.

## More flexibility

This will generate the file at each build, potentially overriding any changes you made to the file. If you want to keep it that way, you can simply remove the `ColorPaletteOverride.json` file from the `Styles` folder, the file won't get overwritten anymore.

> Alternatively, you can also use the [Uno.Dsp.Cli](https://nuget.org/packages/Uno.Dsp.Cli) package to generate the file from the command line. This will allow you to generate the file only when you want to, and not at each build.

Reviewer: @carldebilly, @iurycarlos no changes at all for this part with the latest changes?

Author: Nope, just the file extension.

> [!TIP]

> > Besides **Material Theme Builder Tool** doesn't export anymore **Material Tokens (DSP)** packages, Uno Plugin still supports this file with no restriction.

Reviewer: Not sure if this is a "TIP" here. It would be better to be a "NOTE" I think

Author: Ok!

> > Besides **Material Theme Builder Tool** doesn't export anymore **Material Tokens (DSP)** packages, Uno Plugin still supports this file with no restriction.

Reviewer 3: ```suggestion
> Although the **Material Theme Builder Tool** doesn't export **Material Tokens (DSP)** packages anymore, the Uno Figma Plugin still supports this file format with no restrictions.
```

> > Besides **Material Theme Builder Tool** doesn't export anymore **Material Tokens (DSP)** packages, Uno Plugin still supports this file with no restriction.

Reviewer 2: cant say "with no restrictions", if you have to "make sure to delete the .json file"

> > To be able to import DSP packages just follow the same steps described previously and save the downloaded dsp zip file as `ColorPaletteOverride.zip` in the `Styles` folder of your application project.

Reviewer: ```suggestion
> To be able to import DSP packages just follow the same steps described previously and save the downloaded DSP zip file as `ColorPaletteOverride.zip` in the `Styles` folder of your application project.
```
